src/dtscalibration/io/apsensing.py

```diff
@@ -411,3 +431,219 @@ def metakey(meta, dict_to_parse, prefix):
+            break
+    return match_exists, file
+
+
+def check_if_tra_exists(filepathlist):
+    """
+    Using AP Sensing N4386B both POSC (.xml) export and trace (.tra) export can be used to log measurements. 
+    This function checks, whether both export options were turned on simultaneously. All files .xml and .tra 
+    must be placed in the same directory.
+
+    Parameters
+    ----------
+    filepathlist : list of str
+        List of paths that point the the .xml files
+
+    Notes:
+    ------
+    All files .xml and .tra must be placed in the same directory.
+
+    Returns:
+    --------
+    tra_available : boolean, 
+        True only, when all .xml files have a corresponding .tra file
+    ordered_tra_filepathlist . list of str
+        if tra_available is True: This list contains a list of filepaths for the 
+        .tra file. The list is ordered the same as the input .xml filepath list.
+    """
+   
+    filedir = os.path.dirname(filepathlist[0])
+    # create list of .tra files in directory
+    dir_content=os.listdir(filedir)
+    dir_content.sort()
+    tra_filelist = []
+    for _, file in enumerate(dir_content):
+        if '.tra' in file:
+            tra_filelist.append(file)
+            continue
+    
+    if len(tra_filelist) < len(filepathlist): # early exit, if less .tra than .xml files available
+        return False, []
+
+    tra_available = True # assume it is and prove it wrong if it's not
+    ordered_tra_filepathlist = []
+    for filepath in filepathlist:
+        filename = os.path.basename(filepath)
+        # 14 char long timestamp is the always at the end of AP sensing exported filenames
+        xml_timestamp = filename[-18:-5]
+        tra_exists, tra_filename =  find_corresponding_tra_file(xml_timestamp,tra_filelist)
+        if not tra_exists:
+            tra_available = False
+            break
+        ordered_tra_filepathlist.append(os.path.join(filedir,tra_filename))
+    
+    return tra_available, ordered_tra_filepathlist
+
+def read_single_tra_file(tra_filepath):
+    """
+    Using AP Sensing N4386B both POSC (.xml) export and trace (.tra) export can be used to log measurements. 
+    This function reads the .tra data and appends it to the dask array, which was read from the POSC export (.xml) file.
+
+    .tra files contain different data then the .xml files from POSC export
+        - more metadata
+        - log_ratio and loss(attenuation) calculated by device
+        - PT100 sensor data (optional only if sensors are connnected to device)
+    
+
+    Parameters
+    ----------
+    tra_filepathlist : list of str
+        List of paths that point the the .tra files
+    Notes:
+    ------
+    more metadata could be read from the .tra file and stored in the dask array
+
+    Returns:
+    --------
+    data_dict : dict containing time series data as lists of float
+    reftemps_dict : dict containing PT100 reference as float
+    date_dict : dict containing time data as str
+    """
```

Comment:
I quickly wrote a generic .tra parser:

```py
def parse_numbers(val: str):
    if val == "True":
        return True
    if val == "False":
        return False
    if val.isdigit():
        return int(val)
    try:  # a bit ugly, but sadly there is no string.isfloat() method...
        return float(val)
    except:
        return val


with open("/home/bart/Downloads/tra/C1_SE_CH1_0_5_m20240130141820.tra") as f:
    file = f.readlines()

data = [line for line in file if line != "\n"]  # drops out empty lines

data_dict = {}

current_section = None
for line in data:
    line = line.replace("\n", "")
    if line.startswith("["):
        current_section = line.replace("[", "").replace("]", "")
        data_dict[current_section] = {}
    else:
        content = line.split(";")
        content = [parse_numbers(val) for val in content]
        content_name = content[0]
        if len(content) == 2:
            data_dict[current_section][content_name] = content[1]
        else:
            data_dict[current_section][content_name] = tuple(content[1:])

print(data_dict["Trace.1"]["Date.Year"])
print(data_dict["Trace.1"][1200])

```

This will create a nested dictionary, and could make this function a lot cleaner.

Reply:
thanks! 
I implemented you generic parser and added a test to test_datastore. I'll push that soon.